### PR TITLE
Add GitHub release action workflow using goreleaser

### PR DESCRIPTION
**Pull request by jadolg**
This action will release a new version of jelease when we push a new tag starting with the letter **v** (`'v*'`)

### CODE REVIEW

File: .goreleaser.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+before:
+  hooks:
+    - go mod tidy
+builds:
+  - env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
```
[the remark this answers is not in the document]
Comment (jadolg) [reply .goreleaser.yaml:12]: By default we get all archs
image

[the remark this answers is not in the document]
Comment (jilleJr) [reply .goreleaser.yaml:12]: Sweet!

File: .goreleaser.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
```
[the remark this answers is not in the document]
Comment (jadolg) [reply .goreleaser.yaml:13]: It does, but there's a ticket for that too :smile: 
Let's tackle them separately.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# SPDX-FileCopyrightText: 2022 Risk.Ident GmbH <contact@riskident.com>
+#
+# SPDX-License-Identifier: CC0-1.0
+
+name: goreleaser
+
+on:
+  push:
+    tags:
+      - 'v*'
```
[the remark this answers is not in the document]
Comment (jadolg) [reply .github/workflows/release.yml:10]: Added.

File: .goreleaser.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jadolg) [reply .goreleaser.yaml:12]: Should we release executables for other platforms too?

[the remark this answers is not in the document]
Comment (jilleJr) [reply .goreleaser.yaml:12]: Yea why not

**Pull request merged**